org.uqbar.project.wollok.tests/src/org/uqbar/project/wollok/tests/interpreter/SetTestCase.xtend

```diff
@@ -10,98 +10,96 @@ class SetTestCase extends CollectionTestCase {
+		const unionSet = #{1, 2, 3, 4, 5}
+		
+		assert.equals(aSet, aSet.union(aSet))
+		assert.equals(unionSet, aSet.union(anotherSet))
+		assert.equals(unionSet, anotherSet.union(aSet))
+		'''.test
 	}
 	
 	@Test
 	def void intersectionWithEmptySet() {
 		'''
-		program p {
-			const aSet = #{1, 2, 3}
-			
-			assert.equals(#{}, aSet.intersection(#{}))
-			assert.equals(#{}, #{}.intersection(aSet))
-		}'''.interpretPropagatingErrors
+		const aSet = #{1, 2, 3}
+		
+		assert.equals(#{}, aSet.intersection(#{}))
+		assert.equals(#{}, #{}.intersection(aSet))
+		'''.test
 	}
 	
 	@Test
 	def void intersectionWithNonEmptySets() {
 		'''
-		program p {
-			const aSet = #{1, 2, 3}
-			const anotherSet = #{3, 4, 5}
-			const intersectionSet = #{3}
-			
-			assert.equals(aSet, aSet.intersection(aSet))
-			assert.equals(intersectionSet, aSet.intersection(anotherSet))
-			assert.equals(intersectionSet, anotherSet.intersection(aSet))
-		}'''.interpretPropagatingErrors
+		const aSet = #{1, 2, 3}
+		const anotherSet = #{3, 4, 5}
+		const intersectionSet = #{3}
+		
+		assert.equals(aSet, aSet.intersection(aSet))
+		assert.equals(intersectionSet, aSet.intersection(anotherSet))
+		assert.equals(intersectionSet, anotherSet.intersection(aSet))
+		'''.test
 	}
 
 	@Test
 	def void differenceWithEmptySet() {
 		'''
-		program p {
-			const aSet = #{1, 2, 3}
+		const aSet = #{1, 2, 3}
 			
-			assert.equals(aSet, aSet.difference(#{}))
-			assert.equals(#{}, #{}.difference(aSet))
-		}'''.interpretPropagatingErrors
+		assert.equals(aSet, aSet.difference(#{}))
+		assert.equals(#{}, #{}.difference(aSet))
+		'''.test
 	}
 	
 	@Test
 	def void differenceWithNonEmptySets() {
 		'''
-		program p {
-			const aSet = #{1, 2, 3}
-			const anotherSet = #{3, 4, 5}
-			
-			assert.equals(#{}, aSet.difference(aSet))
-			assert.equals(#{1, 2}, aSet.difference(anotherSet))
-			assert.equals(#{4, 5}, anotherSet.difference(aSet))
-		}'''.interpretPropagatingErrors
+		const aSet = #{1, 2, 3}
+		const anotherSet = #{3, 4, 5}
+		
+		assert.equals(#{}, aSet.difference(aSet))
+		assert.equals(#{1, 2}, aSet.difference(anotherSet))
+		assert.equals(#{4, 5}, anotherSet.difference(aSet))
+		'''.test
 	}
 	
 	@Test
 	def void equalityCaseTrue() {
 		'''
-			program p {
-				assert.equals(#{'Hello'}, #{'Hello'})
-				assert.equals(#{5, 4, 9}, #{4, 5, 9})
-				assert.equals(#{true}, #{true})
-				assert.equals(#{}, #{})
-			}
-		'''.interpretPropagatingErrors		
+		assert.equals(#{'Hello'}, #{'Hello'})
+		assert.equals(#{5, 4, 9}, #{4, 5, 9})
+		assert.equals(#{true}, #{true})
+		assert.equals(#{}, #{})
+		'''.test		
 	}
 			
 	@Test
 	def void equalityCaseFalse() {
 		'''
-			program p {
-				assert.notEquals(#{'Hello'}, #{'Hellou'})
-				assert.notEquals(#{5, 4, 9}, #{4, 5, 3})
-				assert.notEquals(#{4, 5, 9}, [4, 5, 9])
-			}
-		'''.interpretPropagatingErrors		
-	}	
+			assert.notEquals(#{'Hello'}, #{'Hellou'})
+			assert.notEquals(#{5, 4, 9}, #{4, 5, 3})
+			assert.notEquals(#{4, 5, 9}, [4, 5, 9])
```

Comment:
Hi @fdodino , it seems like you have added a lot of tests! Awesome!

But I think you have also used simple, literal objects like strings and numbers for your tests, do you? I'd be nice to add perhaps more complex scenarios, where custom objects are involved, right?